helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/ResourceAssignmentOptimizerAccessor.java

```diff
@@ -277,7 +296,26 @@ private AssignmentResult computeOptimalAssignmentForResources(InputFields inputF
           result);
     }
 
-    return result;
+    if (inputFields.returnFormat.equals(AssignmentFormat.CurrentStateFormat.name())) {
+      return changeToCurrentStateFormat(result);
+    } else {
+      return result;
```

Comment:
Same reason as the comment above, "else" here is too wide considering if more enum items are going to be added. Switch with the default condition would help to avoid potential bugs in the future.

Reply:
And I would suggest letting the changeXXXXXFormat function handle this check. It will just return whatever format requested.